ambari-server/src/main/resources/scripts/post-user-creation-hook.sh

```diff
@@ -82,21 +82,32 @@ echo "Processing post user creation hook payload ..."
 JSON_INPUT="$CSV_FILE.json"
 echo "Generating json file $JSON_INPUT ..."
 
+REALM=${HDFS_PRINCIPAL##*@}
+
 echo "[" | cat > "$JSON_INPUT"
 while read -r LINE
 do
   USR_NAME=$(echo "$LINE" | awk -F, '{print $1}')
+
+  if [ "$SECURITY_TYPE" ==  "KERBEROS" ]
+  then
+    OWNER="${USR_NAME}@${REALM}"
```

Comment:
I believe that this is incorrect and maybe the issue is not really as indicated in the description.  This is what would happen assuming my user kinit-ed as rlevas@EXAMPLE.COM and the auth-to-local rules translated that principal to "rlevas" - which is the default behavior when Ambari manages the auth-to-local rules and the configured realm is EXAMPLE.COM:

```
[rlevas@c7401 ~]$ hdfs dfs -ls /user
Found 4 items
drwxrwx---   - ambari-qa          hdfs          0 2018-08-03 21:27 /user/ambari-qa
drwxrwxr-x   - oozie              hdfs          0 2018-08-03 23:16 /user/oozie
drwx------   - rlevas             hdfs          0 2018-09-06 13:39 /user/rlevas
drwx------   - rlevas@EXAMPLE.COM hdfs          0 2018-09-06 13:43 /user/rlevas_princ
[rlevas@c7401 ~]$ hdfs dfs -ls /user/rlevas_princ
ls: Permission denied: user=rlevas, access=READ_EXECUTE, inode="/user/rlevas_princ":rlevas@EXAMPLE.COM:hdfs:drwx------
```

Reply:
Robert is right, using the REALM, EXAMPLE.COM won't solve all customer scenarios.